integer/src/lib.rs

```diff
@@ -665,6 +665,47 @@ impl_integer_for_usize!(u32, test_integer_u32);
 impl_integer_for_usize!(u64, test_integer_u64);
 impl_integer_for_usize!(usize, test_integer_usize);
 
+/// An iterator over binomial coefficients.
+pub struct BinomialCoeff<T> {
+    a: T,
+    n: T,
+    k: T,
+}
+
+impl<T> BinomialCoeff<T>
+    where T: Integer,
+{
+    /// For a given n, iterate over all binomial coefficients ((k, n - k), binomial(n, k)).
+    pub fn new(n: T) -> BinomialCoeff<T> {
+        BinomialCoeff {
+            k: T::zero(), a: T::one(), n: n
+        }
+    }
+}
+
+impl<T> Iterator for BinomialCoeff<T>
+    where T: Integer + Clone,
+          for<'a> &'a T: std::cmp::PartialEq<&'a T>
+{
+    type Item = ((T, T), T);
+
+    fn next(&mut self) -> Option<((T, T), T)> {
+        if &self.k > &self.n {
+            return None;
+        }
+        self.a = if &self.k != &T::zero() {
+            (self.a.clone() * (self.n.clone() - self.k.clone() + T::one())) / self.k.clone()
```

Comment:
I'm surprised this doesn't need to use your `multiply_and_divide`, as you're basically walking the same series of operations that `binomial` does.